extensions/iam/oauth2/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/spi/JwtDecorator.java

```diff
@@ -0,0 +1,8 @@
+package org.eclipse.dataspaceconnector.iam.oauth2.spi;
+
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jwt.JWTClaimsSet;
+
+public interface JwtDecorator {
```

Comment:
actually it's not the Decorator pattern, so the naming may be misleading - maybe "handler" is a better naming? Nevertheless I must admit I understood it intuitively.

Reply:
Would the term *Writer be more appropriate?

Reply:
maybe *Configurator? but I'm fine with keeping Decorator for now. We can rename it as soon as we find a better name